OCD-4554: Developer statuses only include banned or suspended changes

kekey1 opened the pull request.

kekey1 committed:
feat!: Deprecate developer status history; add status start/end dates

[#OCD-4554]

kekey1 committed:
feat!: Deprecate developer statusEvents in favor of statuses

Statuses include a start date and end date in LocalDate format.

[#OCD-4554]

kekey1 committed:
fix: various issues with developer status changes

[#OCD-4554]

kekey1 committed:
feat: Beginning to mark endpoints as having deprecated response fields

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
feat: finish documenting endpoints with deprecated response fields

[#OCD-4554]

kekey1 committed:
fix: Correctly use new developer statuses field in more places

[#OCD-4554]

kekey1 committed:
style: Move colon to newline

[#OCD-4554]

kekey1 committed:
refactor: Remove unused file

[#OCD-4554]

kekey1 committed:
fix!: Developer ban/email occurs for past or present listing status chg

[#OCD-4574]
[#OCD-4577]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
fix!: Correctly change developer status on listing termination

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
fix: Remove check for two dev statuses in a row

A developer could be banned multiple times during different timeframes.

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
refactor: Change startDay/endDay to -Date fields for dev status

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
feat!: Remove 'showDeleted' request parameter from GET /developers

Breaking Change

[#OCD-4554]

tmy1313 commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/DeveloperStatus.java:
```diff
@@ -22,12 +22,12 @@ public class DeveloperStatus implements Serializable {
+        this.setName(other.getName());
     }
 
     @Override
```
Is there something "special" about these equals and hashcode methods?  Can we just use the standard Lombok ones that are implemented with `@Data`?

tmy1313 commented on chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/DeveloperStatusEvent.java:
```diff
@@ -17,35 +21,34 @@
 
-    // Not all attributes have been included. The attributes being used were selected so the DeveloperManager could
-    // determine equality when updating a Developer
     @Override
```
Same question as above about equals and hashcode methods....

kekey1 committed:
refactor: Remove standard equals/hashcode in favor of lomboks

[#OCD-4554]

kekey1 committed:
fix: Show expected error message if product owner is banned

The prior code showed a helpful error message and something about the
changes in this PR caught the error earlier in the Spring
PermissionsEvaluator classes and was just returning AccessDenied. I
don't really like the change I've done in this commit but I don't
understand how it was NOT returning just Access Denied before or how the
other changes I made resulted in it starting to return that. It should
have always been giving Access Denied. Anyway, the change here results
in the desired behavior for the user so I guess it's fine.

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
feat: Add logging for debugging date overlap

[#OCD-4554]

kekey1 committed:
Revert "feat: Add logging for debugging date overlap"

[#OCD-4554]

This reverts commit 90f6dc89a476b2c515c25f8290d13bae3e3156cb.

kekey1 committed:
fix: Add transactional to DAO method

This method is called from a Permissions class to check whether the
current user can update product ownership. For some reason, in the
Tomcat 9/javax world, this works fine, but in the Tomcat 10/jakarta
world the permissions class does not have a transaction. Annotating the
method works to fix it but I'm not sure if it's the best permanent
solution.

[#OCD-4554]

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
Merge branch 'staging' into OCD-4554

kekey1 committed:
refactor: switch to jakarta based on merge of 4237

[#OCD-4554]
